src/parse-html.js

```diff
@@ -18,58 +18,64 @@ try {
   d.appendChild(parsed)
   if (d.firstChild.classList[0] !== testClass) throw new Error('Parsed classes not preserved')
   supportsHTMLType = true
+} catch (e) {}
+
+var mockDoc = document.implementation.createHTMLDocument('')
+var mockHTML = mockDoc.documentElement
+var mockBody = mockDoc.body
+try {
+  // Check if browser supports documentElement.innerHTML
+  mockHTML.innerHTML += ''
+  supportsInnerHTML = true
 } catch (e) {
-  var mockDoc = document.implementation.createHTMLDocument('')
-  var mockHTML = mockDoc.documentElement
-  var mockBody = mockDoc.body
-  try {
-    // Check if browser supports documentElement.innerHTML
-    mockHTML.innerHTML += ''
-    supportsInnerHTML = true
-  } catch (e) {
-    // Check if browser supports xhtml parsing.
-    parser.parseFromString(testCode, xhtmlType)
-    var bodyReg = /(<body[^>]*>)([\s\S]*)<\/body>/
+  // Check if browser supports xhtml parsing.
+  parser.parseFromString(testCode, xhtmlType)
+  var bodyReg = /(<body[^>]*>)([\s\S]*)<\/body>/
+}
+
+function DOMParserParse(markup, rootName) {
+  var doc = parser.parseFromString(markup, htmlType)
+  // patch for UIWebView not always returning doc.body synchronously
+  if(!doc.body)
```

Comment:
This is what does the trick

Reply:
Thank you for fixing this! Should you submit a PR back to the main repo?